### PR TITLE
Simple Nix derivation for slippc

### PR DESCRIPTION
**Pull request by ramirez7**
The bare minimum to package it with Nix.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,12 @@
+let pkgs = import <nixpkgs> {};
```
Comment (ramirez7) [inline default.nix:1]: By including this, `nix-build` will build the `slippc` binary and output it under a `result/` symlink directory.

If someone wants to add this to their package set, they can `callPackage (import slippcSrc) {};`)

(Laziness + default arguments are fun).

File: default.nix [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+in { stdenv ? pkgs.stdenv, xz ? pkgs.xz }:
+stdenv.mkDerivation {
+  name = "slippc";
+  src = ./.;
+  buildDepends = [ xz ];
```
Comment (ramirez7) [inline default.nix:6]: This is `liblzma`

https://github.com/NixOS/nixpkgs/blob/2b7c73e4a6ad20fac26defc8a657ab2072387fd7/pkgs/tools/compression/xz/default.nix#L88

File: default.nix [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+  installPhase = ''
+    mkdir -p $out
+    cp ./slippc $out/
```
Comment (ramirez7) [inline default.nix]: There's no `make install` target so we have to do this part ourselves.

File: .gitignore [hunk 4]
```diff
@@ -14,3 +14,4 @@ __*
 .todo
 test-replays/zlptest.zlp
 test-replays/zlptest.slp
+result
```
Comment (ramirez7) [inline .gitignore:17]: `nix-build` output

File: default.nix [hunk 5]
```diff
@@ -0,0 +1,12 @@
+let pkgs = import <nixpkgs> {};
+in { stdenv ? pkgs.stdenv, xz ? pkgs.xz }:
+stdenv.mkDerivation {
+  name = "slippc";
+  src = ./.;
+  buildDepends = [ xz ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./slippc $out/bin/
```
Comment (ramirez7) [inline default.nix:10]: Using `bin/` here makes it work with `nix-env -f default.nix` and I think is what Nix expects the `installPhase` to do for executables?

**Comment by pcrain** [comment]
Neat! :open_mouth: Don't have access to a Nix machine myself at the moment, but this certainly isn't gonna break anything, so I'm just gonna trust this works. Thanks a ton! :heart: :smile:

**Pull request merged**